src/msc.cpp

```diff
@@ -89,6 +89,44 @@ static bool convertMapFile(TextStream &t,const QCString &mapName,const QCString
   return true;
 }
 
+bool do_mscgen_generate(const QCString& inFile,const QCString& outFile,mscgen_format_t msc_format,
+                     const QCString &srcFile,int srcLine)
+{
+  QCString const& external_mscgen = Config_getString(MSCGEN);
+  if (!external_mscgen.isEmpty()) {
+    QCString type;
+    switch (msc_format)
+    {
+      case mscgen_format_png:
+        type = "png";
+        break;
+      case mscgen_format_eps:
+        type = "eps";
+        break;
+      case mscgen_format_svg:
+        type = "svg";
+        break;
+      case mscgen_format_pngmap:
+      case mscgen_format_svgmap:
+        type = "ismap";
+        break;
+    }
+    int exitcode = Portable::system(external_mscgen,"-T"+type+" -o "+outFile+" "+inFile);
+    if (exitcode==0)
+      return true;
+    warn(srcFile,srcLine,"Problems running %s given with MSCGEN (exit status: %d); check your installation. Trying with internal mscgen instead.",
+        external_mscgen.data(),exitcode);
+  }
+  int code = mscgen_generate(inFile.data(),outFile.data(),msc_format);
+  if (code!=0)
+  {
```

Comment:
This indicate that when `MSCGEN` is set and the generation of the image fails it attempts to create it with the build in generator? I think that this is unwanted behavior, I think it should be either the external generator or the internal generator but not both.

Reply:
Yes, that was the idea but I can see you point and 

1. Remove this behavior, make the warn into an err, no problem.
2. I can add another config option like MSCGEN_FALLBACK_TO_INTERNAL, and check for it on line 117.

Which would you prefer?

Reply:
No I wouldn't prefer this, the external mscgen should anyway be a superset of the internal function.

Reply:
OK, do you consider this resolved then?